src/Gaufrette/Adapter/AwsS3.php

```diff
@@ -110,8 +110,8 @@ public function rename($sourceKey, $targetKey)
         );
 
         try {
-            $this->service->copyObject($options);
-            return true;
+            $this->service->copyObject(array_merge($options, $this->getMetadata($targetKey)));
```

Comment:
It's been a long time since is issued this patch but if I remember correctly without this call the renamed file was always private no matter what visibility option I used.